feat: use bigdecimal instead of double

tswistak opened the pull request:
<!-- Thank you for improving Fossify. Please consider filling out the details -->

#### Type of change(s)
- [X] Feature / enhancement

#### What changed and why

- Changed number type from Double to BigDecimal, both in calculator and unit converter.
    - For division I've set `DECIMAL128` precision, as `UNLIMITED` was causing errors, e.g., while trying to convert 1 meter to parsecs (however, I don't know who would like to do such conversion `¯\_(ツ)_/¯`).
    - In the unit converter, I've updated some factors to use higher precision numbers.
    - Also, I've found that astronomical units had wrong factor used. I've used one from Wikipedia.
- Used [EvalEx](https://ezylang.github.io/EvalEx/) in place of exp4j, as it supports BigDecimal natively.
    - Thanks to this, implementing https://github.com/FossifyOrg/Calculator/issues/7 should be an easy task, as someone won't need to write all the algorithmic stuff, just provide a string to the library.
    - Also, if you'd consider doing this: https://github.com/FossifyOrg/Calculator/issues/24, there's an extension for EvalEx supporting scientific functions on BigDecimals: https://github.com/ezylang/EvalEx-big-math.
- EvalEx seems to support square roots a bit differently than exp4j, so I had some issues with fixing the current implementation. If there's too much mess around square roots support, I may try to fix it somehow.
    - If you don't mind, I've allowed putting sqrt sign after the number, as it makes it easier to square root a result of a previous calculation. 

I hope that thanks to this PR, Fossify Calculator can come out of the beta 🤞.

BTW. I haven't seen exp4j in 3rd party licenses, and honestly, I don't know how to add there the EvalEx license. Does it have to be done in Commons?

#### Before/After Screenshots/Screen Record


https://github.com/user-attachments/assets/6a7e396f-9212-4c26-939e-d4377d237a96



https://github.com/user-attachments/assets/e4bc3f08-0a63-4303-a2a8-808eeadbcb04



#### Fixes the following issue(s)
<!-- Prefix issues with "Fixes" so that GitHub closes them when the PR is merged (note that each "Fixes #" should be in its own item). -->
- Fixes #6 

#### Checklist
- [X] I read the [contribution guidelines](../blob/HEAD/CONTRIBUTING.md).
- [X] I manually tested my changes on device/emulator (if applicable).
- [X] I updated the "Unreleased" section in `CHANGELOG.md` (if applicable).
- [X] All checks are passing. 

<!-- NOTE: Keep CHANGELOG.md updates clear and concise, they are visible to *all* users. -->

naveensingh reviewed (changes requested):
> DECIMAL128 precision

`MathContext.DECIMAL128` is being accessed in three places. Maybe consider exporting it as `MATH_CONTEXT` [constant](https://github.com/tswistak/Calculator/blob/feat/increase-precision/app/src/main/kotlin/org/fossify/math/helpers/Constants.kt)?

> I've allowed putting sqrt sign after the number, as it makes it easier to square root a result of a previous calculation.

I think it's unconventional. Other Calculator apps just throw a format error in this case. We can keep the original behavior for now. The UX issue will essentially be solved when 'cursor-based editing' is added.

> BTW. I haven't seen exp4j in 3rd party licenses, and honestly, I don't know how to add there the EvalEx license. Does it have to be done in Commons?

Yes, mostly in [Commons](https://github.com/FossifyOrg/commons/blob/1ddccc45d7ec3ee60e64b8312b438d81f4e825e5/commons/src/main/kotlin/org/fossify/commons/activities/LicenseActivity.kt#L34-L68) followed by a minor change in [MainActivity](https://github.com/tswistak/Calculator/blob/33eed811dfaf63ee846bf83397774f886b351322/app/src/main/kotlin/org/fossify/math/activities/MainActivity.kt#L192C24-L192C31). It's a little manual for my taste. Feel free to do it as part of this PR or another (or not), the license list is already outdated. 

---

I flagged two minor issues in the code. Other than that, this looks good :rocket: 

Thank you.

tswistak commented:
> `MathContext.DECIMAL128` is being accessed in three places. Maybe consider exporting it as `MATH_CONTEXT` [constant](https://github.com/tswistak/Calculator/blob/feat/increase-precision/app/src/main/kotlin/org/fossify/math/helpers/Constants.kt)?

That makes sense. Initially, I was experimenting with setting `UNLIMITED` in a few places but ended with `DECIMAL128` everywhere. I'll refactor it.

> I think it's unconventional. Other Calculator apps just throw a format error in this case. We can keep the original behavior for now. The UX issue will essentially be solved when 'cursor-based editing' is added.

Ok, that will be just removing two if-cases (one for solving, other for formatting). I'll do it.

> Yes, mostly in [Commons](https://github.com/FossifyOrg/commons/blob/1ddccc45d7ec3ee60e64b8312b438d81f4e825e5/commons/src/main/kotlin/org/fossify/commons/activities/LicenseActivity.kt#L34-L68) followed by a minor change in [MainActivity](https://github.com/tswistak/Calculator/blob/33eed811dfaf63ee846bf83397774f886b351322/app/src/main/kotlin/org/fossify/math/activities/MainActivity.kt#L192C24-L192C31). It's a little manual for my taste. Feel free to do it as part of this PR or another (or not), the license list is already outdated.

As there's no exp4j now, I think there's no need to rush it. When I'll get to the code, I'll try adding it to Commons and then also to this PR.

> I flagged two minor issues in the code. Other than that, this looks good 🚀

I've already commited one of them. Whitespaces issue I'll later fix in Android Studio, I need to check where there are also such problems.

tswistak commented:
Finished 😄:
- now we have MATH_CONTEXT const to use the same one everywhere
- I reverted how sqrt was working earlier
- I've added EvalEx license to Commons: https://github.com/FossifyOrg/commons/pull/136 and used it here
